mix/qml/TransactionLog.qml

```diff
@@ -6,190 +6,190 @@ import QtQuick.Layouts 1.1
 import org.ethereum.qml.RecordLogEntry 1.0
 
 Item {
+    property ListModel fullModel: ListModel{}
+    property ListModel transactionModel: ListModel{}
+    property ListModel callModel: ListModel{}
 
-	property ListModel fullModel: ListModel{}
-	property ListModel transactionModel: ListModel{}
-	property ListModel callModel: ListModel{}
+    Action {
+        id: addStateAction
+        text: "Add State"
+        shortcut: "Ctrl+Alt+T"
+        enabled: codeModel.hasContract && !clientModel.running;
+        onTriggered: projectModel.stateListModel.addState();
+    }
+    Action {
+        id: editStateAction
+        text: "Edit State"
+        shortcut: "Ctrl+Alt+T"
+        enabled: codeModel.hasContract && !clientModel.running && statesCombo.selectedIndex >= 0 && projectModel.stateListModel.count > 0;
+        onTriggered: projectModel.stateListModel.editState(statesCombo.selectedIndex);
+    }
+    Action {
```

Comment:
We are going to add them in another place in this panel. I will send you
more details by mail.

2015-03-31 14:37 GMT+02:00 Ali Mashatan <notifications@github.com>:

> In mix/qml/TransactionLog.qml
> <https://github.com/ethereum/cpp-ethereum/pull/1472#discussion_r27474436>:
>
> > -	property ListModel callModel: ListModel{}
> > +    Action {
> > +        id: addStateAction
> > +        text: "Add State"
> > +        shortcut: "Ctrl+Alt+T"
> > +        enabled: codeModel.hasContract && !clientModel.running;
> > +        onTriggered: projectModel.stateListModel.addState();
> > +    }
> > +    Action {
> > +        id: editStateAction
> > +        text: "Edit State"
> > +        shortcut: "Ctrl+Alt+T"
> > +        enabled: codeModel.hasContract && !clientModel.running && statesCombo.selectedIndex >= 0 && projectModel.stateListModel.count > 0;
> > +        onTriggered: projectModel.stateListModel.editState(statesCombo.selectedIndex);
> > +    }
> > +    Action {
>
> what about "playAndPauseAction" and play/pause icon ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ethereum/cpp-ethereum/pull/1472/files#r27474436>.
>